core/common/connector-core/src/main/java/org/eclipse/edc/connector/core/store/ContainsOperatorPredicate.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.core.store;
+
+import org.eclipse.edc.spi.query.OperatorPredicate;
+
+import java.util.Collection;
+
+public class ContainsOperatorPredicate implements OperatorPredicate {
+
+    @Override
+    public boolean test(Object property, Object operandRight) {
+        if (property instanceof Collection<?> collection) {
+            return collection.contains(operandRight);
```

Comment:
theoretically `operandRight` could be a `List` itself, IDK what would happen then...

Reply:
in the (postgre)SQL context it can be only a single object, because eventually I guess another operator should be used.
I think if we want to being able to pass a list as `operandRight` maybe we'll need another operator (`containsAll`?)